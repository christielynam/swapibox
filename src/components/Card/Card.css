.card-style {
  width: 200px;
  height: 450px;
  margin: 20px;
  background-color: whitesmoke;
}

img {
  width: 35px;
  height: 35px;
  cursor: pointer
}

.favorited {
  color: #81EC7C
}