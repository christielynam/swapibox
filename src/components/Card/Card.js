import React, { Component } from 'react'
import styled from 'styled-components'
import lightsaber from '../../assets/lightsaber.png'
import './Card.css'

const CardStyle = styled.div`
  width: 200px;
  height: 450px;
  margin: 20px;
  background-color: whitesmoke;
  img {
    width: 35px;
    height: 35px;
    cursor: pointer
  }
`

const Card = ({ item, toggleFavorite }) => {
  return (
    <CardStyle onClick={() => toggleFavorite(item.type, item.name)}>
      <img src={lightsaber} alt='lightsaber icon - favorite' />
      <h3 className={item.favorited ? 'favorited' : ''}>{item.name}</h3>
      {item.homeworld && <p>Homeworld: {item.homeworld}</p>}
      {item.population && <p>Population: {item.population}</p>}
      {item.species && <p>Species: {item.species}</p>}
      {item.terrain && <p>Terrain: {item.terrain}</p>}
      {item.climate && <p>Climate: {item.climate}</p>}
      {item.residents && <p>Residents: {item.residents.toString()}</p>}
      {item.model && <p>Model: {item.model}</p>}
      {item.class && <p>Class: {item.class}</p>}
      {item.passengers && <p>Passengers: {item.passengers}</p>}
    </CardStyle>
  )  
}

export default Card