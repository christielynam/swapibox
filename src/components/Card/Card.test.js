import React from 'react';
import Card from './Card';
import { CardStyle } from './Card';
import { shallow } from 'enzyme'

describe('Card', () => {
  let wrapper
  let mockItem
  let mockToggleFavorite

  beforeEach(() => {
    mockItem = {name: 'Christie', homeworld: 'Earth', population: '500', species: 'Human', terrain: 'mountains', climate: 'rainy', residents: ['Hudson', 'Bradley', 'Missy'], model: 'Range Rover', class: 'HSE Sport', passengers: '5'}
    mockToggleFavorite = jest.fn()
    wrapper = shallow(<Card item={mockItem} toggleFavorite={mockToggleFavorite} />)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call toggleFavorite when the card is clicked', () => {
    const card = wrapper.find(CardStyle)

    card.simulate('click')

    expect(mockToggleFavorite).toHaveBeenCalled()
  })
  
})