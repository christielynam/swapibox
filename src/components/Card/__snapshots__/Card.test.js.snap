// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
<styled.div
  className="card-style"
  onClick={[Function]}
>
  <img
    alt="lightsaber icon - favorite"
    className="light-saber"
    src="lightsaber.png"
  />
  <h3
    className=""
  >
    Christie
  </h3>
  <p>
    Homeworld: 
    Earth
  </p>
  <p>
    Population: 
    500
  </p>
  <p>
    Species: 
    Human
  </p>
  <p>
    Terrain: 
    mountains
  </p>
  <p>
    Climate: 
    rainy
  </p>
  <p>
    Residents: 
    Hudson,Bradley,Missy
  </p>
  <p>
    Model: 
    Range Rover
  </p>
  <p>
    Class: 
    HSE Sport
  </p>
  <p>
    Passengers: 
    5
  </p>
</styled.div>
`;
